src/features/usersSlice.ts

```diff
@@ -0,0 +1,51 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { User } from '../types/User';
+import { client } from '../utils/fetchClient';
+
+export interface UsersState {
+  items: User[];
+  loaded: boolean;
+  hasError: boolean;
+}
+
+const initialState: UsersState = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+// Асинхронный thunk для загрузки пользователей
+export const loadUsers = createAsyncThunk('users/load', async () => {
+  const response = await client.get<User[]>('/users');
+
+  return response;
+});
+
+const usersSlice = createSlice({
+  name: 'users',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder
+      .addCase(loadUsers.pending, state => {
+        // eslint-disable-next-line no-param-reassign
+        state.loaded = false;
+        // eslint-disable-next-line no-param-reassign
+        state.hasError = false;
+      })
+      .addCase(loadUsers.fulfilled, (state, action) => {
+        // eslint-disable-next-line no-param-reassign
+        state.items = action.payload;
+        // eslint-disable-next-line no-param-reassign
+        state.loaded = true;
+      })
+      .addCase(loadUsers.rejected, state => {
+        // eslint-disable-next-line no-param-reassign
+        state.hasError = true;
+        // eslint-disable-next-line no-param-reassign
+        state.loaded = true;
+      });
+  },
+});
+
+export default usersSlice.reducer;
```

Comment:
It's a good practice to export the actions created by `createSlice`. This allows you to dispatch these actions in your components. You can do this by adding `export const { ... } = usersSlice.actions;`.